src/pages/api/breadcrumbs.ts

```diff
@@ -0,0 +1,74 @@
+import apiUrl from '../../utils/apiUrl';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+type QueryObject = {[key: string]: string | string[]}
+type ReturnedData = {[key: string]: {[key: string]: string}}
+
+export default async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const returnedData = await fetchData(req.query);
+    const breadcrumbs = pathToCrumbs(req.query.path as string, returnedData);
+    res.status(200).json({ breadcrumbs });
+};
+
+const pathToCrumbs = (path: string, data: ReturnedData) => {
+    const regexp = new RegExp(/\[(.*?)\]/);
```

Comment:
It pains my programmer heart to say this, but I don't think a regex is the right solution for this use case. It's much more readable to just do `beginsWith('[') && endsWith(']')` for comparisons, and `slice(1, -1)` to remove the brackets and get at the name within.

I would probably have done the same way you did, with a regex, but now when I'm on the receiving end of this pull request it actually took me a couple of minutes to even understand what this did.

That could be remedied with different naming, and perhaps wrapping it in a function with a clear name, but in the end that's actually way more code than just doing the slightly more verbose comparisons using common string methods.